lite2/client.go

```diff
@@ -138,103 +138,6 @@ type Client struct {
 	logger log.Logger
 }
 
-// NewClient returns a new light client. It returns an error if it fails to
-// obtain the header & vals from the primary or they are invalid (e.g. trust
-// hash does not match with the one from the header).
-//
-// Witnesses are providers, which will be used for cross-checking the primary
-// provider. At least one witness must be given. A witness can become a primary
-// iff the current primary is unavailable.
-//
-// See all Option(s) for the additional configuration.
-func NewClient(
```

Comment:
I don't think we should move these methods

Reply:
can you explain why

Reply:
it's common to have a struct and constructors in one file